frontend/pages/ManageControlsPage/OSUpdates/components/WindowsTargetForm/WindowsTargetForm.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useContext, useState } from "react";
+import { isEmpty } from "lodash";
+import classnames from "classnames";
+
+import { APP_CONTEXT_NO_TEAM_ID } from "interfaces/team";
+import { NotificationContext } from "context/notification";
+import configAPI from "services/entities/config";
+import teamsAPI from "services/entities/teams";
+
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import Button from "components/buttons/Button";
+import validatePresence from "components/forms/validators/validate_presence";
+
+const baseClass = "windows-target-form";
+
+interface IWindowsTargetFormData {
+  deadlineDays: string;
+  gracePeriodDays: string;
+}
+
+interface IWindowsTargetFormErrors {
+  deadlineDays?: string;
+  gracePeriodDays?: string;
+}
+
+// validates that a string is a number from 0 to 30
+const validateDeadlineDays = (value: string) => {
+  if (value === "") return false;
+
+  const parsedValue = Number(value);
+  return Number.isInteger(parsedValue) && parsedValue >= 0 && parsedValue <= 30;
+};
+
+// validates string is a number from 0 to 7
+const validateGracePeriodDays = (value: string) => {
+  if (value === "") return false;
+
+  const parsedValue = Number(value);
+  return Number.isInteger(parsedValue) && parsedValue >= 0 && parsedValue <= 7;
+};
+
+const validateForm = (formData: IWindowsTargetFormData) => {
+  const errors: IWindowsTargetFormErrors = {};
+
+  if (!validatePresence(formData.deadlineDays)) {
+    errors.deadlineDays = "The deadline days is required.";
+  } else if (!validateDeadlineDays(formData.deadlineDays)) {
+    errors.deadlineDays = "Deadline must meet criteria below.";
+  }
+
+  if (!validatePresence(formData.gracePeriodDays)) {
+    errors.gracePeriodDays = "The grace period days is required.";
+  } else if (!validateGracePeriodDays(formData.gracePeriodDays)) {
+    errors.gracePeriodDays = "Grace period must meet criteria below.";
+  }
+
+  return errors;
+};
+
+const createMdmConfigData = (deadlineDays: string, gracePeriodDays: string) => {
```

Comment:
Maybe you could type the return value as something like `Pick<IMdmConfig, "windows_updates">`? Renaming the function `newMdmConfigWindowsUpdates` could also help signal that it is generating an instance of a defined object.